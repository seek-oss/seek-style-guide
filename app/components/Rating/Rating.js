import styles from './Rating.less';

import React, { Component } from 'react';
import Baseline from 'react-baseline';

import GridContainer from 'GridContainer/GridContainer';
import SandboxPreview from 'SandboxPreview/SandboxPreview';
import SandboxToggle from 'SandboxToggle/SandboxToggle';
import Section from 'Section/Section';
import HeadlineText from 'HeadlineText/HeadlineText';
import Spec from 'Spec/Spec';
import Code from 'Code/Code';

import SeekRating from 'seek-style-guide/react/Rating/Rating';

const specs = {
  default: {
    Height: '2 grid rows'
  }
};

const getSpec = specsObj => {
  return Object.keys(specsObj).reduce((result, specName) => ({
    ...result,
    ...(specsObj[specName] ? specs[specName] : {})
  }), {});
};

export default class RatingDemo extends Component {
  constructor() {
    super();

    this.state = {
      baseline: false
    };

    this.toggleBaseline = this.toggleBaseline.bind(this);
  }

  toggleBaseline(event) {
    this.setState({
      baseline: event.target.checked
    });
  }

  render() {
    const { baseline } = this.state;

    const spec = getSpec({
      default: true
    });

    const rating = (<SeekRating rating={2.6} />);

    return (
      <div>
        <Baseline isVisible={baseline} color="#eee">
          <div className={styles.sandboxContainer}>
            <GridContainer>
              <div className={styles.sandbox}>
                <SandboxPreview>
                  {rating}
                </SandboxPreview>
                <div style={{ position: 'absolute', top: 0, right: 0 }}>
                  <SandboxToggle
                    label="Baseline"
                    toggleType="checkbox"
                    toggleProps={{
                      type: 'checkbox',
                      checked: baseline,
                      onChange: this.toggleBaseline
                    }}
                  />
                </div>
              </div>
            </GridContainer>
          </div>
        </Baseline>

        <GridContainer className={styles.gridContainer}>
          <Section className={styles.section}>
            <HeadlineText>Spec</HeadlineText>
            <Spec spec={spec} />
          </Section>

          <Section className={styles.section}>
            <HeadlineText>Code</HeadlineText>
            <Code jsx={rating} />
          </Section>
        </GridContainer>
      </div>
    );
  }
}
