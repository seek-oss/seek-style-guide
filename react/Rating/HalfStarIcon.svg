<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M505.1 177.5c-2.8-6.4-9.1-10.6-16.1-10.6l-148.4-.3-67.9-147.3C269.8 13 263.6 9 256.7 9h-.1c-6.9.1-13.2 4.2-15.9 10.5l-64.8 147.1-155.4.3c-7.1 0-13.5 4.3-16.2 10.8-2.7 6.5-1.2 14.1 3.7 19.1l121.7 122.8-36.8 158.9c-1.6 6.9 1.1 14 6.8 18.2 3.1 2.2 6.7 3.4 10.3 3.4 3.1 0 6.2-.8 9-2.5L256.8 415 398 497.6c6 3.5 13.5 3.1 19.2-.9 5.6-4.1 8.3-11.1 6.8-17.9l-35.1-161.2 112.7-121.1c5-5.2 6.2-12.6 3.5-19zm-148 122.8c-3.9 4.2-5.5 10.1-4.3 15.7l28.5 131.1-115.7-67.7c-2.5-1.5-5.4-2.3-8.2-2.4V70.3l55.9 121.1c2.9 6.2 9.1 10.2 15.9 10.3l119.4.3-91.5 98.3z"/></svg>
