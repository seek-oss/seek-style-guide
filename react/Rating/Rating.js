import styles from './Rating.less';

import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import StarIcon from '../StarIcon/StarIcon';

import ScreenReaderOnly from '../ScreenReaderOnly/ScreenReaderOnly';
import Text from '../Text/Text';

import withTextProps, { SizePropTypes } from '../private/withTextProps';

const getPercent = (rating, position) =>
  Math.round(Math.min(Math.max(rating - position, 0), 1) * 100);

const getStar = (percent, key, starClassName, size) => {
  const props = {
    key,
    size,
    className: styles.star,
    svgClassName: starClassName,
    filled: percent >= 75,
    half: percent >= 25 && percent < 75,
    _large: true
  };

  return <StarIcon {...props} />;
};

const Rating = ({
  description,
  rating,
  starClassName,
  showTextRating,
  size,
  ...restProps
}) => {
  const extendedStarClassName = classnames({ [starClassName]: starClassName });

  return (
    <Text raw size={size} regular {...restProps}>
      <ScreenReaderOnly>{rating} out of 5</ScreenReaderOnly>
      <span className={styles.rating}>
        {[...Array(5)].map((v, position) => {
          const percent = getPercent(rating, position);
          return getStar(percent, position, extendedStarClassName, size);
        })}
        {(showTextRating || Boolean(description)) && (
          <span className={styles.textRating}>
            {rating.toFixed(1)}
            {description ? ' ' : ''}
            {description}
          </span>
        )}
      </span>
    </Text>
  );
};

Rating.displayName = 'Rating';

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
  className: PropTypes.string,
  starClassName: PropTypes.string,
  showTextRating: PropTypes.bool,
  description: PropTypes.any,
  ...SizePropTypes
};

Rating.defaultProps = {
  showTextRating: false,
  size: 'standard'
};

export default withTextProps(Rating);
