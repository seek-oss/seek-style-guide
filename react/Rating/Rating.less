@import (reference) "~seek-style-guide/theme";

@elements-spacing: 0.2em;

.rating {
  display: flex;
  align-items: flex-end;
}

.star {
  color: @sk-orange;
  display: inline;
  margin-right: @elements-spacing;

  &:last-child {
    margin-right: 0;
  }
}

.textRating {
  margin-left: @elements-spacing;
}
