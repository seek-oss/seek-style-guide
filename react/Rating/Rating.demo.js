import Rating from './Rating';
import React from 'react';

const Description = () => <span>overall rating</span>;

export default {
  route: '/rating',
  title: 'Rating',
  category: 'Typography',
  component: Rating,
  initialProps: {
    hero: true,
    rating: 0
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Show rating text',
          transformProps: ({ ...props }) => ({
            ...props,
            showTextRating: true
          })
        },
        {
          label: 'Show rating text with description',
          transformProps: ({ ...props }) => ({
            ...props,
            showTextRating: true,
            description: <Description />
          })
        }
      ]
    },
    {
      label: 'Ratings',
      type: 'radio',
      states: [0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5].map(rating => ({
        label: `${String(rating)} Stars`,
        transformProps: props => ({
          ...props,
          rating
        })
      }))
    },
    {
      label: 'Size',
      type: 'radio',
      states: ['Hero', 'Headline', 'Heading', 'Large', 'Standard', 'Small'].map(
        size => ({
          label: `${String(size)}`,
          transformProps: ({ hero, ...props }) => ({
            ...props,
            [size.toLowerCase()]: true
          })
        })
      )
    }
  ]
};
