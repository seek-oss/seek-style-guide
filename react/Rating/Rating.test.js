import { render } from 'enzyme';
import React from 'react';
import Rating from './Rating';

describe('Rating', () => {
  it('should render with starClassName', () => {
    const wrapper = render(
      <Rating rating={5.0} starClassName={'star-class-name'} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with className', () => {
    const wrapper = render(
      <Rating rating={5.0} className={'root-class-name'} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with description', () => {
    const wrapper = render(
      <Rating rating={5.0} description="overall rating from..." />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with description as a component', () => {
    const SomeComponent = () => <span>overall rating</span>;
    const wrapper = render(
      <Rating rating={5.0} description={<SomeComponent />} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('should render correct star rating', () => {
    const ratings = [2.74, 2.75, 3.74, 3.75, 0.0, 5.0];

    ratings.forEach(rating => {
      it(`when rating is: ${rating}`, () => {
        const wrapper = render(<Rating rating={rating} />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('should render correct size', () => {
    const sizes = ['heading', 'headline', 'hero'];

    sizes.forEach(size => {
      it(`when size is: ${size}`, () => {
        const wrapper = render(<Rating rating={3.5} {...{ [size]: true }} />);
        expect(wrapper).toMatchSnapshot();
      });
    });

    sizes.forEach(size => {
      it(`when size is: ${size} and rating text is shown`, () => {
        const wrapper = render(
          <Rating rating={4} {...{ [size]: true }} showTextRating />
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
