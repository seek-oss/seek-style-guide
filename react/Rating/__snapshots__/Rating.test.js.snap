// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render correct size when size is: heading 1`] = `
<span
  class="Text__root Text__heading Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      3.5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__heading Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct size when size is: heading and rating text is shown 1`] = `
<span
  class="Text__root Text__heading Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      4 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__heading Rating__star"
      >
        mock svg
      </span>
      <span
        class="Rating__textRating"
      >
        4.0
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct size when size is: headline 1`] = `
<span
  class="Text__root Text__headline Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      3.5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__headline Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct size when size is: headline and rating text is shown 1`] = `
<span
  class="Text__root Text__headline Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      4 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__headline Rating__star"
      >
        mock svg
      </span>
      <span
        class="Rating__textRating"
      >
        4.0
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct size when size is: hero 1`] = `
<span
  class="Text__root Text__hero Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      3.5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__hero Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct size when size is: hero and rating text is shown 1`] = `
<span
  class="Text__root Text__hero Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      4 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__hero Rating__star"
      >
        mock svg
      </span>
      <span
        class="Rating__textRating"
      >
        4.0
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct star rating when rating is: 0 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      0 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct star rating when rating is: 2.74 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      2.74 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct star rating when rating is: 2.75 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      2.75 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct star rating when rating is: 3.74 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      3.74 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct star rating when rating is: 3.75 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      3.75 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render correct star rating when rating is: 5 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render with className 1`] = `
<span
  class="Text__root root-class-name Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render with description 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Rating__textRating"
      >
        5.0 overall rating from...
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render with description as a component 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Rating__textRating"
      >
        5.0 
        <span>
          overall rating
        </span>
      </span>
    </span>
  </span>
</span>
`;

exports[`Rating should render with starClassName 1`] = `
<span
  class="Text__root Text__standard Text__raw Text__baseline"
>
  <span
    class="Regular__root"
  >
    <span
      class="ScreenReaderOnly__root"
    >
      5 out of 5
    </span>
    <span
      class="Rating__rating"
    >
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
      <span
        class="Icon__root StarIcon___large StarIcon__filled StarIcon__standard Rating__star"
      >
        mock svg
      </span>
    </span>
  </span>
</span>
`;
